internal/controller/util.go

```diff
@@ -69,3 +72,62 @@ func isStatefulSetReady(ctx context.Context, c client.Client, name, namespace st
 
 	return false, nil
 }
+
+func waitForFullSync(ctx context.Context, c client.Client, pod *corev1.Pod, maxDuration time.Duration) error {
+	ctx, cancel := context.WithTimeout(ctx, maxDuration)
+	defer cancel()
+	for {
+		select {
+		case <-ctx.Done():
+			return fmt.Errorf("timed out waiting for full sync")
+		default:
+			// Check if the statefulset is ready
+			ready, err := isFullSync(ctx, c, pod)
+			if err != nil {
+				return err
+			}
+			if ready {
+				return nil
+			}
+		}
+	}
+}
+
+func isFullSync(ctx context.Context, c client.Client, pod *corev1.Pod) (bool, error) {
```

Comment:
It is not clear from the name if isFullSync means that full sync is in progress or done.